### PR TITLE
Fix issue with DiffIndex on initial commit

### LINKED ISSUE #11650: [BUG] When view first commit it return "Diff Content Not Available"
- Gitea version (or commit ref):  1.13.0+dev-67-g24be06d7a
- Git version:
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL) https://try.gitea.io/LJR/SuperProejeto/commit/98120fc540b39a4bb7be70fa8802867d03f09bb6
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

When you view the first commit of a repo. It returns "Diff Content Not Available".
This error occur on https://try.gitea.io/. And the https://gitea.com/ everything works well.

After reviewing the recent commit history，The error may be related to this pr #11614 

## Screenshots
image

### PR DESCRIPTION
**Pull request by zeripath**
Unfortunately #11614 introduced a bug whereby the initial commit of a
repository could not be seen due to there being no parent commit to
create a clear diff from.

Here we create a diffstat from the difference between the parentless SHA and the SHA of the empty tree - a constant known to git. (With thanks to @L0veSunshine for informing me of this SHA)

Thanks to @a1012112796 for initial attempt to fix.

Fix #11650

Closes #11674 

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by zeripath** [comment]
As ever with this part of the codebase no good deed goes unpunished. 

I've said it before and no doubt I'll say it again the compare page endpoint is fragile, convoluted and highly inefficient.

It needs a complete refactor.

**Review by lafriks — APPROVED** [review]

File: services/gitdiff/gitdiff.go [hunk 1]
```diff
@@ -711,7 +712,21 @@ func GetDiffRangeWithWhitespaceBehavior(repoPath, beforeCommitID, afterCommitID
 		return nil, fmt.Errorf("Wait: %v", err)
 	}
 
-	diff.NumFiles, diff.TotalAddition, diff.TotalDeletion, err = git.GetDiffShortStat(repoPath, beforeCommitID+"..."+afterCommitID)
+	shortstatArgs := []string{beforeCommitID + "..." + afterCommitID}
+	var env []string
+	if len(beforeCommitID) == 0 || beforeCommitID == git.EmptySHA {
+		shortstatArgs = []string{"--cached", afterCommitID}
```
Comment (zeripath) [inline services/gitdiff/gitdiff.go]:
```suggestion
		shortstatArgs = []string{"--cached", "-R", afterCommitID}
```

**Comment by a1012112796** [comment]
please review #11674 again, Thanks

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Comment by a1012112796** [comment]
Ok, I see. I thought in the wrong direction. Some nits for this change. 
First, as we know, the reason is that the ``ParsePatch`` does not count the real number of files but counts the number of files to be shown, so if the file number is more than max file number, it will only return it. but if the file number is less than max file number, it won't be necessary to count again. For example, 
```DIFF
diff --git a/services/gitdiff/gitdiff.go b/services/gitdiff/gitdiff.go
index 02aef7088..7d5e31084 100644
--- a/services/gitdiff/gitdiff.go
+++ b/services/gitdiff/gitdiff.go
@@ -711,6 +711,10 @@ func GetDiffRangeWithWhitespaceBehavior(repoPath, beforeCommitID, afterCommitID
 		return nil, fmt.Errorf("Wait: %v", err)
 	}
 
+	if !diff.IsIncomplete {
+		return diff, err
+	}
+
 	shortstatArgs := []string{beforeCommitID + "..." + afterCommitID}
 	if len(beforeCommitID) == 0 || beforeCommitID == git.EmptySHA {
 		shortstatArgs = []string{git.EmptyTreeSHA, afterCommitID}
```

Then, I still wonder whether it's necessary to show the real number of files if the file number is more than max file number. Maybe showing a plus sign is enough, for example ``100+``, if users want to know more, they can increase ``git.MAX_GIT_DIFF_FILES`` when they have a good internet connection.

**Pull request merged**